bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/DownloadServlet.java

```diff
@@ -95,16 +110,59 @@ private boolean checkForInlineSelector(SlingHttpServletRequest request) {
         return bytes;
     }
 
+    @SuppressWarnings("UnstableApiUsage")
     private void sendResponse(byte[] bytes, Asset asset, SlingHttpServletResponse response, boolean inline) throws IOException {
         response.setContentType(asset.getMimeType());
         response.setContentLength(bytes.length);
         if (inline) {
-            response.setHeader(CONTENT_DISPOSITION, "inline");
+            response.setHeader(CONTENT_DISPOSITION_HEADER, "inline");
         } else {
-            response.setHeader(CONTENT_DISPOSITION, "attachment; filename=\"" + asset.getName() + "\"");
+            response.setHeader(CONTENT_DISPOSITION_HEADER, "attachment; filename=\"" + asset.getName() + "\"");
         }
+        response.setHeader("ETag", ByteSource.wrap(bytes).hash(Hashing.md5()).toString().toUpperCase());
+        response.setHeader("Last-Modified", getLastModifiedDate(asset));
```

Comment:
This is available as constant below https://sling.apache.org/apidocs/sling11/org/apache/sling/api/servlets/HttpConstants.html.